### PR TITLE
Place /stats routes behind login check, for performance reasons

### PR DESCRIPTION
**Pull request by jywarren**
await approval in #9002, but this is a pretty simple change!

### CODE REVIEW

**Comment by jywarren** [comment]
This probably broke some tests which expect to be able to get to `/stats` -- should be a relatively easy fix!

**Comment by jywarren** [comment]
Yes - changes needed on:

*         test/functional/stats_controller_test.rb:30:in `block in <class:StatsControllerTest>'
*         test/integration/public_pages_test.rb:80:in `block in <class:PublicPagesTest>'

**Comment by jywarren** [comment]
```
Progress: |======== FAIL I18nTest#test_should_choose_i18n_for_notes/stats (200.61s)
        Expected at least 1 element matching "h2", found 0..
        Expected 0 to be >= 1.
        test/integration/I18n_test.rb:509:in `block (2 levels) in <class:I18nTest>'
        test/integration/I18n_test.rb:500:in `each'
        test/integration/I18n_test.rb:500:in `block in <class:I18nTest>'
```

```
Progress: |====== FAIL StatsControllerTest#test_should_assign_correct_value_to_graph_comments_on_GET_stats (21.89s)
        --- expected
        +++ actual
        @@ -1 +1 @@
        -nil
        +{1617148800000.0=>0, 1617753600000.0=>0, 1618358400000.0=>0, 1618963200000.0=>0, 1619568000000.0=>0}
        test/functional/stats_controller_test.rb:33:in `block in <class:StatsControllerTest>'

 FAIL StatsControllerTest#test_should_load_stats_range_query (22.04s)
        Expected response to be a <2XX: success>, but was a <302: Found> redirect to <http://test.host/login?return_to=/stats>
        Response body: <html><body>You are being <a href="http://test.host/login?return_to=/stats">redirected</a>.</body></html>
        test/functional/stats_controller_test.rb:40:in `block in <class:StatsControllerTest>'
```

File: test/functional/stats_controller_test.rb [hunk 1]
```diff
@@ -32,6 +35,7 @@ def setup
   end
 
   test 'should load stats range query' do
+    UserSession.create(users(:bob)) # we now require login for this page for load reasons
     get :index
     assert_response :success
```
Comment (jywarren) [inline test/functional/stats_controller_test.rb:40]: why should this be a 302 redirect???

**Comment by cesswairimu** [comment]
Hi Jeff, 
I think I found the prob. The integration test that was failing after inspecting I found that it was returning an flash message `You must be logged in to access this page`You must be logged in to access this page` adding a session on the test solved it.

The functional one  error was ` Authlogic::Session::Activation::NotActivatedError: You must activate the Authlogic::Session::Base.controller with a controller object before creating objects` fixed by adding `activate_authlogic` under setup to activate the sessions created in the tests 

hopefully we are all green now  :crossed_fingers:  thanks

**Pull request merged**